### PR TITLE
feat: add name as a filter for ListStores

### PR DESCRIPTION
**Pull request by kalleep**
## Description
While the this [rfc](https://github.com/openfga/rfcs/pull/27) is not yet accepted its still useful to implement the ability to filter stores by name for ListStores requests. 

## References
Part of https://github.com/openfga/rfcs/pull/27

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

If you haven't done so yet, we would appreciate it if you could star the [OpenFGA repository](https://github.com/openfga/openfga). :)

### CODE REVIEW

**Comment by kalleep** [comment]
I have have a branch with an implementation for this for openfga server. I will open up a draft pr with it and link them.

File: openfga/v1/openfga_service.proto [hunk 1]
```diff
@@ -1687,6 +1687,8 @@ message ListStoresRequest {
     (validate.rules).string.max_bytes = 5120,
     (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"eyJwayI6IkxBVEVTVF9OU0NPTkZJR19hdXRoMHN0b3JlIiwic2siOiIxem1qbXF3MWZLZExTcUoyN01MdTdqTjh0cWgifQ==\""}
   ];
+	
+  string name = 3 [(google.api.field_behavior) = OPTIONAL];
```
[the remark this answers is not in the document]
Comment (kalleep) [reply openfga/v1/openfga_service.proto:1691]: Oh yeah that makes sense :+1:

File: openfga/v1/openfga_service.proto [hunk 2]
```diff
@@ -1687,6 +1687,15 @@ message ListStoresRequest {
     (validate.rules).string.max_bytes = 5120,
     (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"eyJwayI6IkxBVEVTVF9OU0NPTkZJR19hdXRoMHN0b3JlIiwic2siOiIxem1qbXF3MWZLZExTcUoyN01MdTdqTjh0cWgifQ==\""}
   ];
+
+  string name = 3 [
+    (google.api.field_behavior) = OPTIONAL,
```
Comment (rhamzeh) [inline openfga/v1/openfga_service.proto:1692]: I think `OPTIONAL` is the default and doesn't need to be stated.

That said, it doesn't hurt to have it specified.

File: openfga/v1/openfga_service.proto [hunk 3]
```diff
@@ -1687,6 +1687,15 @@ message ListStoresRequest {
...
+    (validate.rules).string = {
+      pattern: "^[a-zA-Z0-9\\s\\.\\-\\/^_&@]{3,64}$"
+      ignore_empty: true
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"my-store-name\""}
```
Comment (rhamzeh) [inline openfga/v1/openfga_service.proto:1697]: Can we add a description here?

Something along the lines of:
`The name parameter instructs the API to only include results that match that name. Multiple results may be returned. Only exact matches will be returned; substring matches and regexes will not be evaluated`

File: openfga/v1/openfga_service.proto (hunk 2 above)
Comment (kalleep) [reply openfga/v1/openfga_service.proto:1692]: Yeah the default is optional but I think it better to rely on explicit behavior instead of implicit

**Review by jpadilla — DISMISSED** [review]

**Review by rhamzeh — APPROVED** [review]

**Review by jpadilla — APPROVED** [review]

**Comment by jpadilla** [comment]
@kalleep we had to fix something else in https://github.com/openfga/api/pull/214 mind updating again?

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**